makelib/golang.mk

```diff
@@ -73,14 +78,17 @@ ifneq ($(GO_TEST_SUITE),)
 GO_TEST_FLAGS += -run '$(GO_TEST_SUITE)'
 endif
 
-GOPATH := $(shell go env GOPATH)
+# no GOPATH is set, since go modules are being used
+GOPATH := 
+
+# enable go modules
+export GO111MODULE=on
 
 # setup tools used during the build
-DEP_VERSION=v0.5.1
-DEP := $(TOOLS_HOST_DIR)/dep-$(DEP_VERSION)
 GOJUNIT := $(TOOLS_HOST_DIR)/go-junit-report
 GOCOVER_COBERTURA := $(TOOLS_HOST_DIR)/gocover-cobertura
 GOIMPORTS := $(TOOLS_HOST_DIR)/goimports
+CONTROLLERGEN := $(TOOLS_HOST_DIR)/controller-gen
```

Comment:
Not sure the `golang.mk` is the right place to have `controller-gen` and `make manifests`. They seem a little more specific so maybe merit their own Makefile? Maybe we have a `k8scontrollers.mk` or something?

Reply:
I was thinking about that too. The issue is that `go.generate` depends on `controller-gen` too. Do you think it would make sense to move `go.generate` out of this makefile as well?

Reply:
hmm I don't think that having `controller-gen` in `go.generate` is necessary as we didn't previously. I would probably remove `manifests` though.

Reply:
I changed the current go generate tagging in the go code so that it points to the controller-gen binary, instead of running its `main.go` from vendor folder

https://github.com/crossplaneio/stack-aws/pull/20/commits/22c2fc211187c96dfb26653977dff5a89fb96379#diff-3f58b311e6905ae34e04e497070f4881R18

Reply:
added `k8s_controllers.mk`

Reply:
can this be removed from there then? Also have you seen `kubebuilder.mk`? seems like it overlaps with `k8s_controller.mk` -- I'd recommend combining.